chart/compass/charts/system-broker/templates/deployment.yaml

```diff
@@ -58,10 +58,11 @@ spec:
             # - name: SERVER_SHUTDOWN_TIMEOUT
             #   value: "0.0.0.0:{{ .Values.deployment.args.containerPort }}"
             - name: SERVER_SELF_URL
-              value: "https://{{ .Values.global.system_broker.tls.host }}.{{ .Values.global.ingress.domainName }}/broker"
-            - name: GRAPHQL_CLIENT_GRAPHQL_ENDPOINT
+              value: "https://{{ .Values.global.gateway.mtls.host }}.{{ .Values.global.ingress.domainName }}"
+            - name: GRAPHQL_CLIENT_GRAPHQL_ENDPOINT # directly call the compass-gateway, as we already have a crafted jwt id token
+              value: "http://{{ .Values.global.gateway.tls.host }}.{{ .Release.Namespace }}.svc.cluster.local:{{.Values.global.gateway.port }}/director/graphql"
+            - name: GRAPHQL_CLIENT_GRAPHQL_OAUTH_ENDPOINT # used when system-broker acts as an integration system, we need to convert the oauth2 opaque access token to jwt id token
               value: "https://{{ .Values.global.gateway.tls.secure.oauth.host }}.{{ .Values.global.ingress.domainName }}/director/graphql"
-
             # - name: GRAPHQL_CLIENT_ENABLE_LOGGING
```

Comment:
Why are all these configurations left out and commented? We should leave out dead code/configuration.

Reply:
Yes, we should avoid dead code or dead configurations but it's okay to leave them for now just for the sake of being a template. They can be removed/uncommented when we reach the productization stage.